src/pkg/packager/deploy.go

```diff
@@ -714,21 +721,26 @@ func (p *Packager) installChartAndManifests(ctx context.Context, componentPaths
 	return installedCharts, nil
 }
 
-func (p *Packager) printTablesForDeployment(ctx context.Context, componentsToDeploy []types.DeployedComponent) {
+func (p *Packager) printTablesForDeployment(ctx context.Context, componentsToDeploy []types.DeployedComponent) error {
 	// If not init config, print the application connection table
 	if !p.cfg.Pkg.IsInitConfig() {
 		message.PrintConnectStringTable(p.connectStrings)
-	} else {
-		if p.cluster != nil {
-			// Grab a fresh copy of the state (if we are able) to print the most up-to-date version of the creds
-			freshState, err := p.cluster.LoadZarfState(ctx)
-			if err != nil {
-				freshState = p.state
-			}
-			// otherwise, print the init config connection and passwords
-			message.PrintCredentialTable(freshState, componentsToDeploy)
-		}
+		return nil
+	}
+	// Don't print if cluster is not configured
+	if p.cluster == nil {
+		return nil
+	}
+	// Grab a fresh copy of the state to print the most up-to-date version of the creds
+	latestState, err := p.cluster.LoadZarfState(ctx)
+	if err != nil {
+		return err
 	}
+	if latestState == nil {
+		latestState = p.state
+	}
```

Comment:
I think we can delete this. We only had it because we kept going even if we errored.

Reply:
Yeah here we set the latestState if it's nil, but it's only going to be nil if it error'd and now we return on error.